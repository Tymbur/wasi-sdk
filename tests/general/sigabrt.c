#include <signal.h>
#include <stdio.h>
#include <stdlib.h>

int main(void) {
    fprintf(stderr, "raising SIGABRT...\n");
    raise(SIGABRT);
    fprintf(stderr, "oops!\n");
    return EXIT_FAILURE;
}
