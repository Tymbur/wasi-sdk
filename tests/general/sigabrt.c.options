-D_WASI_EMULATED_SIGNAL -lwasi-emulated-signal
