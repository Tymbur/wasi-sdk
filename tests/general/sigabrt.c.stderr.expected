raising SIGABRT...
Program recieved fatal signal: Aborted
Error: failed to run main module `sigabrt.c.---.wasm`

Caused by:
    0: failed to invoke `_start`
    1: wasm trap: unreachable, source location: @----
       wasm backtrace:
